File: stack/stack.cpp
#include<bits/stdc++.h>
using namespace std;

struct Mystack
{
    int *arr;
    int cap;
    int top;
    Mystack(int c)
    {
        cap=c;
        arr=new int [c];
        top =-1;
    }
void push(int x)
{   if(top<cap){
    top++;
    arr[top]=x;}
    else{cout<<"overflow";}
}
int pop()
{   if(top>-1){
    int res =arr[top];
    top--;
    return res;
}
else {cout<<"underflow";}
}
int peek()
{
    return arr[top];
}
int size()
{
    return(top+1);
}
bool isEmpty()
{
    return (top==-1);
}
};



//for dynamic testing use vector 
// use push back and pop back.

File: stack/stack.java
import java.util.*;
 

class Mystack
{
    int arr[];
    int cap;
    int top;
    Mystack(int c)
    {
        top=-1;
        cap=c;
        arr=new int[cap];
    }
    void push(int x)
    {
        top++;
        arr[top]=x;
    }
    int pop()
    {
        int res=arr[top];
        top--;
        return res;
    }
    int size()
    {
        return top+1;
    }
    boolean isempty()
    {
        return(top==-1);
    }
}


//for dynamic allocation just use Array list and create handle cases



// can also be implemented using link list 